### PR TITLE
python: fix FluxExecutorFuture cancellation bug

### PR DESCRIPTION
**Pull request by jameshcorbett**
A bugfix and a behavioral change that I'm happy to split if you think it would be better.

The bug is that calling ``Future.cancel()`` more than once will raise an error. I've already been bitten by this more than once.

The behavior change is that when a future is cancelled, calling ``Future.jobid()`` will now raise a ``CancelledError`` rather than returning -1.

### CODE REVIEW

File: src/bindings/python/flux/job/executor.py [hunk 1]
```diff
@@ -193,16 +193,22 @@ def jobid(self, timeout=None):
...
+        :return: a positive integer jobid.
 
         :raises concurrent.futures.TimeoutError: If the jobid is not available
             before the given timeout.
+        :raises concurrent.futures.CancelledError: If the future was cancelled.
         """
         if self.__jobid is not None:
+            # if the future is cancelled the jobid is set to an exception
+            if isinstance(self.__jobid, concurrent.futures.CancelledError):
+                raise self.__jobid
```
Comment (grondo) [inline src/bindings/python/flux/job/executor.py:205]: Would it be clearer to just `raise CancelledError` here? The current approach seems fine as well (though it is confusing the LGTM check as well). I guess I'm curious if there's extra context in the `self.__jobid` object that prevents you from raising a `CancelledError` directly. (reminder: I'm not all that familiar with Python and its idioms)

Comment (jameshcorbett) [reply src/bindings/python/flux/job/executor.py:205]: Yeah I think that would probably be better. There's no extra context that I know of. I could also change the ``if isinstance()`` check to ``if self.cancelled()`` which is probably clearer.

**Comment by grondo** [comment]
Unfortunately you have to rerun all the workflow jobs (click on the Details link next to the check and it will take you to the GitHub Actions page for the PR. There's a drop down button on the right that lets you re-run the entire workflow) I've just restarted the CI.

**Review by grondo — APPROVED** [review]
LGTM! Thanks!!

**Comment by jameshcorbett** [comment]
I forgot to tag @andre-merzky to check with him whether the -1/CancelledError change is all right with him. Referencing https://github.com/ExaWorks/psi-j-python/pull/14

**Comment by andre-merzky** [comment]
> I forgot to tag @andre-merzky to check with him whether the -1/CancelledError change is all right with him. Referencing [ExaWorks/psi-j-python#14](https://github.com/ExaWorks/psi-j-python/pull/14)

Thanks for the ping @jameshcorbett - that looks sensible to me!

**Comment by jameshcorbett** [comment]
Great, then I think this is good to merge.

**Comment by grondo** [comment]
Excellent. I've set the MWP flag.

**Pull request merged**